lib/containers/bats.pm

```diff
@@ -336,11 +348,52 @@ sub bats_patches {
     return if get_var("BATS_URL");
 
     my $package = get_required_var("BATS_PACKAGE");
+    $package = ($package eq "aardvark") ? "aardvark-dns" : $package;
+
     my $github_org = ($package eq "runc") ? "opencontainers" : "containers";
 
     foreach my $patch (split(/\s+/, get_var("BATS_PATCHES", ""))) {
-        my $url = "https://github.com/$github_org/$package/pull/$patch.diff";
+        my $url = ($patch =~ /^\d+$/) ? "https://github.com/$github_org/$package/pull/$patch.diff" : $patch;
         record_info("patch", $url);
         assert_script_run "curl -sL $url | patch -p1 --merge";
     }
 }
+
+sub bats_sources {
+    my $version = shift;
+
+    my $package = get_required_var("BATS_PACKAGE");
+    $package = ($package eq "aardvark") ? "aardvark-dns" : $package;
+
+    my $github_org = ($package eq "runc") ? "opencontainers" : "containers";
+    my $tag = "v$version";
+
+    # Support these cases for BATS_URL:
+    # 1. As full URL: https://github.com/containers/aardvark-dns/archive/refs/heads/main.tar.gz
+    # 2. As GITHUB_ORG#TAG: SUSE#suse-v4.9.5, yourusername#test-patch, etc
+    # 3. As TAG only: main, v1.2.3, cool-test-fix, etc
+    # 4. Empty. Use default for repo based on package version
+
+    my $url = get_var("BATS_URL", "");
+    if ($url !~ m%^https://%) {
```

Comment:
How does this behave when `$url == ""` ? I suppose this condition is true, therefore `$url` will be updated on line #385 accordingly. Is this the case 4. "Empty. Use default for repo based on package version" ?

Reply:
> How does this behave when `$url == ""` ? I suppose this condition is true, therefore `$url` will be updated on line #385 accordingly. Is this the case 4. "Empty. Use default for repo based on package version" ?

Yes.